engine.go

```diff
@@ -138,7 +144,35 @@ func (eng *Engine) ClockAt(name string, start time.Time, tags ...Tag) *Clock {
 	}
 }
 
+var GoVersionReportingEnabled = os.Getenv("STATS_DISABLE_GO_VERSION_REPORTING") != "true"
+
 func (eng *Engine) measure(t time.Time, name string, value interface{}, ftype FieldType, tags ...Tag) {
+	if GoVersionReportingEnabled {
+		eng.once.Do(func() {
+			vsn := strings.TrimPrefix(runtime.Version(), "go")
+			parts := strings.Split(vsn, ".")
+			// older Go version might be "go1.13"
+			if len(parts) == 2 || len(parts) == 3 {
+				maj, err := strconv.Atoi(parts[0])
+				if err == nil {
+					eng.measure_(t, "go_version.major", maj, Gauge)
+				}
+				min, err := strconv.Atoi(parts[1])
+				if err == nil {
+					eng.measure_(t, "go_version.minor", min, Gauge)
+				}
+				if len(parts) == 3 {
+					patch, err := strconv.Atoi(parts[2])
+					if err == nil {
+						eng.measure_(t, "go_version.patch", patch, Gauge)
+					}
+				}
+			}
```

Comment:
Would it be sufficient to just ship the version string (but keep that trimmed prefix, good looking out) and let folks on the consumer side manage/mangle/parse the value as they see fit? I'm not sure Major or Minor have much utility in isolation.

Reply:
I didn't see a way to report a string instead of a number, but I will look again

Reply:
Yeah fair, I don't know the stats stack intimately so maybe it's just not a thing. Thanks for checking.

Reply:
What if we put the version string in a tag, and set a gauge/counter to `1`? Then you can sum by version tag.